storage/jobs.go

```diff
@@ -0,0 +1,31 @@
+package storage
+
+import "github.com/onflow/flow-go/module"
+
+type Job interface {
+	// each job has a unique ID for deduplication
+	ID() module.JobID
+}
+
+// Jobs is the reader for an ordered job queue. Job can be fetched by the index,
+// which start from 0
+type Jobs interface {
+	AtIndex(index int64) (Job, error)
```

Comment:
maybe using uint64 might enforce start from zero

Reply:
I thought about this, I reason I didn't use is because the job queue has a value `ProcessedIndex`, if we force `uint64` then the first index will start from 0, but since the first job hasn't been processed, the `ProcessedIndex` will have to be `-1`, which underflows. 
I just chose int64 to keep it flexible